### ISSUE TITLE
Expand `prefer-dom-node-text-content` against `.innerHTML` or implement `no-inner-html` rule

### ISSUE THREAD

**Issue by darkred** [sindresorhus/eslint-plugin-unicorn#1405]
My suggestion is to expand the [`prefer-dom-node-text-content`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-dom-node-text-content.md) rule to also enforce the use of `.textContent` over `.innerHTML`.
That is because, in extension development such of Refined GitHub,
the use `innerHTML` should always be avoided, otherwise the extension becomes flagged by the extension stores.

This came up in this PR comment https://github.com/sindresorhus/refined-github/pull/4520#discussion_r659362368
As described in the comment above, such a rule is already implemented as a plugin, `eslint-plugin-lwc` with its [`no-inner-html`](https://github.com/salesforce/eslint-plugin-lwc/blob/master/docs/rules/no-inner-html.md).
However, that rule disallows the use of 'innerHTML' in ALL its forms. This includes `innerHTML`, `outputHTML` and `insertAdjacentHTML`

Concluding, I was wondering whether it could be beneficial for XO,
to either expand the existing `prefer-dom-node-text-content` rule against `innerHTML`,
or incorporate that rule `no-inner-html` as a whole.

## Fail

```js
foo.innerHTML = '🦄';
and maybe also
foo.outerHTML = '🦄';
foo.insertAdjacentHTML = '🦄';

```

## Pass

```js
foo.textContent = '🦄';
```

**Comment by sindresorhus** [comment]
Maybe we could make a more general `no-unsafe-dom` rule.

For example, with [Trusted Types](https://github.com/w3c/webappsec-trusted-types), `.innerHTML` is not dangerous.

Alternatively, a `prefer-trusted-types` rule.

**Comment by fregante** [comment]
Related developments: https://web.dev/sanitizer/

```js
❌ el.innerHTML = html;
✅ el.setHTML(html, sanitizer)
